### ISSUE TITLE
[bug]: Textual Inversion batching adds time

### ISSUE THREAD

**Issue by mikeage** [invoke-ai/InvokeAI#3007]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Linux

### GPU

cuda

### VRAM

16GB

### What version did you experience this issue on?

2.3.2

### What happened?

It appears that the time required to run textual inversion is proportional to batch size, rather than having a large batch size run fewer, but longer, iterations. 

(I'm just getting started with TI).

My environment is Debian 11 on an AWS EC2 machine, type g4dn.xlarge, with a Tesla T4 card.

I downloaded 5 images from  https://github.com/huggingface/diffusers/tree/main/examples/dreambooth (of the dog), and resized them to be less than 512 bytes on the longest side). I then ran the following two commands:

```
invokeai-ti --model=v2-1_512-ema-pruned --resolution=512 --learnable_property=object --initializer_token='*' --placeholder_token='<resized_alvan>' --train_data_dir=/home/admin/invokeai/text-inversion-training-data/resized_alvan --output_dir=/mnt/ephemeral/resized_alvan --scale_lr --train_batch_size=4 --gradient_accumulation_steps=4 --max_train_steps=3000 --learning_rate=0.0005 --lr_scheduler=constant --mixed_precision=fp16 --only_save_embeds
```
vs
```
invokeai-ti --model=v2-1_512-ema-pruned --resolution=512 --learnable_property=object --initializer_token='*' --placeholder_token='<resized_alvan>' --train_data_dir=/home/admin/invokeai/text-inversion-training-data/resized_alvan --output_dir=/mnt/ephemeral/resized_alvan --scale_lr --train_batch_size=1 --gradient_accumulation_steps=4 --max_train_steps=3000 --learning_rate=0.0005 --lr_scheduler=constant --mixed_precision=fp16 --only_save_embeds
```

After the first ten steps were completed (out of 3000), the first was estimating approximately 3:50 remaining, whereas the second, 1:07. My expectation was that with a larger batch size, it should allow it to take advantage of parallelization and would be *faster*, up to the point where the GPU was saturated, where it'd start to take about the same amount of time, as it'd be doing fewer cycles that were longer, as the batches got bigger (plus/minus some coordination and partitioning). Instead, as the batch size went up from 1 to 2 to 4, the time grew roughly linearly. 

### Additional context

Some discussion at https://discord.com/channels/1020123559063990373/1088122033449205840/1088122033449205840

### Contact Details

@mikeage on discord, or here

**Comment by lstein** [comment]
This is unexpected. Could you try turning off `--scale_lr` and see what effect this has? Also, I've added instructions on how to make use of multiple GPUs for distributed learning, in case your environment is similar. The docs are in PR #3015

**Comment by mikeage** [comment]
Without `--scale_lr`, this is what I see after five minutes (I left in `--lr_scheduler=constant`, afterwards, I tried without that as well, and it didn't really make any difference)

```
admin@ip-172-31-65-138:~$ invokeai-ti --model=v2-1_512-ema-pruned --resolution=512 --learnable_property=object --initializer_token='*' --placeholder_token='<resized_alvan>' --train_data_dir=/home/admin/invokeai/text-inversion-training-data/resized_alvan --output_dir=/mnt/ephemeral/resized_alvan --train_batch_size=4 --gradient_accumulation_s
teps=4 --max_train_steps=3000 --learning_rate=0.0005 --lr_scheduler=constant --mixed_precision=fp16 --only_save_em
beds
/home/admin/.local/lib/python3.9/site-packages/accelerate/accelerator.py:243: FutureWarning: `logging_dir` is deprecated and will be removed in version 0.18.0 of 🤗 Accelerate. Use `project_dir` instead.
  warnings.warn(
03/24/2023 05:00:58 - INFO - ldm.invoke.training.textual_inversion_training - Distributed environment: NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: fp16

{'clip_sample_range', 'variance_type'} was not found in config. Values will be initialized to default values.
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training - ***** Running training *****
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Num examples = 500
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Num Epochs = 94
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Instantaneous batch size per device = 4
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Total train batch size (w. parallel, distributed & accumulation) = 16
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Gradient Accumulation steps = 4
03/24/2023 05:01:40 - INFO - ldm.invoke.training.textual_inversion_training -   Total optimization steps = 3000
Steps:   2%|█                                            | 73/3000 [05:02<3:22:22,  4.15s/it, loss=0.12, lr=0.008]^C
```

It doesn't appear to have improved anything (I reran the first command, and my initial comment of 3:40, above, really should be closer to 3:30, after a few minutes), I'm afraid.

**Comment by lstein** [comment]
I haven't had time to investigate this, unfortunately. Will try to do so this later this week.

**Comment by mikeage** [comment]
No worries, I'm here if/when you need anything else!

**Comment by mikeage** [comment]
@hipsterusername , do you know when this was fixed? I just tried in 3.0.1, and I'm seeing the exact same thing

**Comment by mikeage** [comment]
If I want a total number of steps of, say, 3000, and I have a batch of 8, is it possible the answer is as simple as setting the `--max_train_steps` parameter to 3000/8 = 375?
